src/gen_alpha_poc/generalized_alpha_time_integrator.cpp

```diff
@@ -272,44 +267,49 @@ std::tuple<State, Kokkos::View<double*>> GeneralizedAlphaTimeIntegrator::AlphaSt
         "Newton-Raphson iterations failed to converge on a solution after " +
         std::to_string(n_iterations + 1) + " iterations!\n"
     );
+
     return results;
 }
 
-Kokkos::View<double*> GeneralizedAlphaTimeIntegrator::UpdateGeneralizedCoordinates(
-    const Kokkos::View<double*> gen_coords, const Kokkos::View<double*> delta_gen_coords
+void GeneralizedAlphaTimeIntegrator::UpdateGeneralizedCoordinates(
+    Kokkos::View<const double* [kNumberOfLieGroupComponents]> gen_coords,
+    Kokkos::View<const double* [kNumberOfLieAlgebraComponents]> delta_gen_coords,
+    Kokkos::View<double* [kNumberOfLieGroupComponents]> gen_coords_next
 ) {
-    auto gen_coords_next = Kokkos::View<double*>("generalized_coordinates_next", gen_coords.size());
     const auto h = this->time_stepper_.GetTimeStep();
 
-    auto update_generalized_coordinates = KOKKOS_LAMBDA(size_t) {
+    // Loop over all nodes in the system and update the generalized coordinates
+    for (size_t node = 0; node < gen_coords.extent(0); node++) {
         // {gen_coords_next} = {gen_coords} + h * {delta_gen_coords}
         //
         // Step 1: R^3 update, done with vector addition
-        auto current_position = Vector{gen_coords(0), gen_coords(1), gen_coords(2)};
+        auto current_position =
+            Vector{gen_coords(node, 0), gen_coords(node, 1), gen_coords(node, 2)};
```

Comment:
We can't access `gen_coords` on host (it lives in device memory).  This existing error is fixed in my run on cuda pull request, so we'll need to resolve this conflicting change in a similar manner to how I did it over here.

Reply:
I seem to have been confusing this with other, similar changes I made.  The code, as it is on main, works.  However, the new code will not run on device, because you moved this code outside of a `KOKKOS_LAMBDA` (line 284 in the old code).  

The easiest way to fix it is to revert back to calling all of this inside a `KOKKOS_LAMBDA`.  Just change your indexing technique to the new approach and then have the parallel_for go over `gen_coords.extent(0)` instead of `1`.